NSArray+F.m

```diff
@@ -84,6 +84,20 @@ + (NSArray *) arrayFrom:(NSInteger) from To:(NSInteger) to {
     return [NSArray arrayWithArray:mutArr];
 }
 
+- (id) flatten {
+    NSMutableArray *mtArr = [NSMutableArray array];
+    for (NSArray *a in self) {
+        if ([a conformsToProtocol:@protocol(NSFastEnumeration)]) {
+            for (id object in [a flatten]) {
+                [mtArr addObject:object];
```

Comment:
My original thought was that testing for NSFastEnumeration might also match NSSet etc; but as NSSet does not have a `flatten` method it fails anyway. I have now adjusted the test to check for the selector `flatten` instead of the NSFastEnumeration protocol, and have changed the iteration as you suggested. If NSSet at some point gets a `flatten` method that would presumably return an NSArray, so all should be well in any case.